sdk/wallet-adapter/adapters/wallet-standard-adapter/src/StandardWalletAdapter.ts

```diff
@@ -2,21 +2,42 @@
 // SPDX-License-Identifier: Apache-2.0
 
 import { SignableTransaction } from "@mysten/sui.js";
-import { WalletAdapter } from "@mysten/wallet-adapter-base";
+import {
+  WalletAdapter,
+  WalletAdapterEvents,
+} from "@mysten/wallet-adapter-base";
 import { StandardWalletAdapterWallet } from "@mysten/wallet-standard";
+import mitt from "mitt";
 
 export interface StandardWalletAdapterConfig {
   wallet: StandardWalletAdapterWallet;
 }
 
+type WalletAdapterEventsMap = {
+  [E in keyof WalletAdapterEvents]: Parameters<WalletAdapterEvents[E]>[0];
+};
+
 export class StandardWalletAdapter implements WalletAdapter {
   connected = false;
   connecting = false;
 
+  readonly #events = mitt<WalletAdapterEventsMap>();
   #wallet: StandardWalletAdapterWallet;
 
   constructor({ wallet }: StandardWalletAdapterConfig) {
     this.#wallet = wallet;
+    this.#wallet.features["standard:events"].on(
```

Comment:
I'm not convinced this should be listened for here vs in wallet kit. I'm curious why you ended up listening here?

Also, if we do keep the listen here, it probably makes more sense to set this up in `connect` instead of the constructor?

Reply:
I had to do it to update the connected field. But also `WalletAdapter` interface seems to kind of isolate the wallet from wallet-kit, and thought maybe it's better to add the extra `WalletAdapterEvents` so adapters can have extra events instead of just exposing the wallet feature. But happy to change it if you have something else in mind.

> Also, if we do keep the listen here, it probably makes more sense to set this up in connect instead of the constructor?

I will move it